File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Ruby/checkbox_test.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class CheckBoxTest < Test::Unit::TestCase
	def setup
		@driver = Selenium::WebDriver.for :chrome
		@driver.get 'http://dl.dropbox.com/u/55228056/Config.html'
		@verification_errors = []
	end
	
	def test_checkbox
		#Get the Checkbox as WebElement using it's value attribute
		airbags = @driver.find_element :xpath => "//input[@value='Airbags']"
				
		#Check if its already selected? otherwise select the Checkbox
 		#by calling click() method
 		if airbags.selected? == false
 			airbags.click
		end
		
		#Verify Checkbox is Selected	
		assert_equal true, airbags.selected?
 		
 		#Check Checkbox if selected? If yes, deselect it
 		#by calling click() method
 		if  airbags.selected?
 			airbags.click
		end
 		
 		#Verify Checkbox is Deselected
 		assert_equal false, airbags.selected?
	end
	
	def teardown
		@driver.quit
		assert_equal [], @verification_errors
	end
end

File: XX - Backup Project/xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Python/checkbox_test.py
from selenium import webdriver
import time, unittest

class CheckBoxTest (unittest.TestCase):
	def setUp(self)	:
		self.driver = webdriver.Chrome()
		
	def test_checkbox(self):
		driver = self.driver
		driver.get("http://dl.dropbox.com/u/55228056/Config.html")
		
		#Get the Checkbox as WebElement using it's value attribute
		airbags = driver.find_element_by_xpath("//input[@value='Airbags']")
		
		#Check if its already selected? otherwise select the Checkbox
 		#by calling click() method
 		if airbags.is_selected() == False:
 			airbags.click()
		
		#Verify Checkbox is Selected	
		self.assertEqual(True,airbags.is_selected())
		
 		#Check Checkbox if selected? If yes, deselect it
 		#by calling click() method
 		if airbags.is_selected():
 			airbags.click()
 		
 		#Verify Checkbox is Deselected
		self.assertEqual(False,airbags.is_selected())
		
	def tearDown(self):
		self.driver.close()
		
if __name__ == "__main__":
	unittest.main()